VSRAD.DebugServer/Server.cs

```diff
@@ -49,19 +61,77 @@ public async Task LoopAsync()
             while (true)
             {
                 var client = await _listener.AcceptTcpClientAsync().ConfigureAwait(false);
-                ClientConnected(client, clientsCount);
+                TcpClientConnected(client, clientsCount);
                 clientsCount++;
             }
         }
 
-        private void ClientConnected(TcpClient tcpClient, uint clientId)
+        private void TcpClientConnected(TcpClient tcpClient, uint clientId)
         {
             var networkClient = new NetworkClient(tcpClient, clientId);
             var clientLog = new ClientLogger(clientId, _verboseLogging);
+            if (!Task.Run(() => TryProcessServerHandshake(tcpClient, clientLog)).Result)
+            {
+                clientLog.HandshakeFailed(networkClient.EndPoint);
+                return;
+            }
             clientLog.ConnectionEstablished(networkClient.EndPoint);
             Task.Run(() => BeginClientLoopAsync(networkClient, clientLog));
         }
 
+        private async Task<bool> TryProcessServerHandshake(TcpClient client, ClientLogger clientLog)
+        {
+            try
+            {
+                StreamWriter writer = new StreamWriter(client.GetStream(), Encoding.UTF8) { AutoFlush = true };
+                StreamReader reader = new StreamReader(client.GetStream(), Encoding.UTF8);
```

Comment:
In this project (at least in Package and Server, not so sure about Syntax) we tend not to use explicit types where possible, so here and below (`clientResponse`) I would definitely go with `var` keyword

Reply:
done